### PR TITLE
feat #355 Allow disable on list item

### PR DESCRIPTION
**Pull request by krose422**
Allows disabling of listItem and exposes disabledStyle prop for custom disabled style to listItem.

### CODE REVIEW

**Comment by kyler-hyuna** [comment]
@krose422 You have been on fire lately :D Been looking at this for a while. Can you happen to do the same for some of the other components? I guess button would be an easy one to do

**Comment by iRoachie** [comment]
@krose422 You have been on fire lately :D Been looking at this for a while. Can you happen to do the same for some of the other components? I guess button would be an easy one to do

File: src/list/ListItem.js [hunk 1]
```diff
@@ -97,8 +99,14 @@ const ListItem = props => {
     <Component
       onLongPress={onLongPress}
       onPress={onPress}
+      disabled={disabled}
       underlayColor={underlayColor}
-      style={[styles.container, containerStyle && containerStyle]}
+      style={[
+        styles.container,
+        containerStyle && containerStyle,
+        disabled && { opacity: .5 },
```
Comment (iRoachie) [inline src/list/ListItem.js]: Could you move this opacity into a style object at the bottom?

**Comment by krose422** [comment]
@iRoachie - Moved that style object. And yeah, I'd love to do the same for others! I thought button already had a disabled option; does it not? Let me know what other components should allow disabling though and I can certainly jump on them. :) Thanks!

**Comment by iRoachie** [comment]
Alright one more thing to pass the tests. Run `npm test -- -u` @krose422

**Comment by kyler-hyuna** [comment]
Sweet thanks again! I think the FormInput doesn't have one

**Pull request merged**